ql/src/java/org/apache/hadoop/hive/ql/QueryState.java

```diff
@@ -221,6 +247,8 @@ public QueryState build() {
       if (generateNewQueryId) {
         String queryId = QueryPlan.makeQueryId();
         queryConf.setVar(HiveConf.ConfVars.HIVEQUERYID, queryId);
+        setMapReduceJobTag(queryConf, queryId);
```

Comment:
Who reads the tag (queryId) set in queryConf? Will tasks read it? If not, then this won't have any effect.

Reply:
semantic analyzer reads it